src/effects.c

```diff
@@ -3009,6 +3012,41 @@ static void HMIWidgetsSetUnit(LV2_HMI_WidgetControl_Handle handle,
     pthread_mutex_unlock(&g_hmi_mutex);
 }
 
+static void HMIWidgetsSetIndicator(LV2_HMI_WidgetControl_Handle handle,
+                                   LV2_HMI_Addressing addressing,
+                                   float indicator_poss)
+{
+    if (handle == NULL || addressing == NULL || g_hmi_data == NULL) {
+        return;
+    }
+
+    const int64_t addressing_check = (int64_t)addressing;
+
+    if (addressing_check < 0x8000 || addressing_check > 0x9000) {
+        return;
+    }
+
+    const int assignment_id = addressing_check - 0x8000;
+
+    if (g_verbose_debug) {
+        printf("DEBUG: HMIWidgetsSetIndicator %i: %f\n", assignment_id, indicator_poss);
+        fflush(stdout);
+    }
+
+    if (indicator_poss < 0.0)
```

Comment:
this should be 0.0f otherwise you are promoting `indicator_poss` to a double before doing the comparison